### PR TITLE
chore: remove stylesheetConfig option

### LINKED ISSUE #3614: Deprecate `customProperties` config from `@lwc/style-compiler`
### Description

Now that LWC has dropped support for compat, we should be able to remove the `customProperties` configuration from the `@lwc/style-compiler` package. This should greatly simplify style generation logic and also enable us to remove the [custom serialization](https://github.com/salesforce/lwc/blob/e93fd757e27e1bb5a32bdccb8bdf3529a3923656/packages/%40lwc/style-compiler/src/serialize.ts) logic.

### PR DESCRIPTION
**Pull request by jmsjtu**
## Details
This deprecates the `stylesheetConfig` option now that LWC has dropped support for IE11 and older browsers.

The deprecation removes the functionality and produces a warning when the `stylesheetConfig` is used.

This resolves #3614 and a follow up PR will need to be done to fulfill #3266.

## Does this pull request introduce a breaking change?

* ✅ No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

* ✅ No, it does not introduce an observable change.

## GUS work item
W-13889465

### CODE REVIEW

File: packages/@lwc/compiler/src/options.ts [hunk 1]
```diff
@@ -139,6 +148,15 @@ function validateOptions(options: TransformOptions) {
...
 
+    if (!isUndefined(options.stylesheetConfig) && !alreadyWarnedOnStylesheetConfig) {
+        alreadyWarnedOnStylesheetConfig = true;
+
+        // eslint-disable-next-line no-console
+        console.warn(
+            `"stylesheetConfig" property is deprecated. The value doesn't impact the compilation and can safely be removed.`
+        );
+    }
+
```
Comment (nolanlawson) [inline packages/@lwc/compiler/src/options.ts:159]: We can combine this `if` condition with the one below it

**Review by nolanlawson — APPROVED** [review]
:shipit:

File: packages/@lwc/compiler/src/options.ts (hunk 1 above)
Comment (jmsjtu) [reply packages/@lwc/compiler/src/options.ts:159]: @nolanlawson updated, removed the `validateStylesheetConfig` as it doesn't really serve any purpose anymore.

Comment (nolanlawson) [reply packages/@lwc/compiler/src/options.ts:159]: Nice! More red. 🎉

**Pull request merged**